Bug 1863000: deployment of vmware fails if URL contains space

glekner opened the pull request.

glekner commented:
@suomiy thoughts?

yaacov commented on frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:
```diff
@@ -18,14 +18,25 @@ export const getV2VVMwareImportProvidersTabValidity = (
   const hasSelectedVM = !!iGetVMWareFieldValue(state, id, VMWareProviderField.VM);
   const hasLoadedVM = !!iGetVMWareFieldAttribute(state, id, VMWareProviderField.VM, 'vm');
+  // check if host name has any kind of whitespace
+  const isHostnameValid = !/\s/.test(iGetVMWareFieldValue(state, id, VMWareProviderField.HOSTNAME));
```
the BZ is just about spaces at the end, am I missing something ?

glekner replied:
why not check for any spaces while we're here? :)

yaacov replied:
can we truncate the hostname ?

suomiy commented on frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/providers/v2vvmware-tab-validation.ts:
```diff
@@ -18,14 +18,25 @@ export const getV2VVMwareImportProvidersTabValidity = (
 
   const hasSelectedVM = !!iGetVMWareFieldValue(state, id, VMWareProviderField.VM);
   const hasLoadedVM = !!iGetVMWareFieldAttribute(state, id, VMWareProviderField.VM, 'vm');
+  // check if host name has any kind of whitespace
```
this logic should be in a new function called validateV2VVMwareImportProvidersTab. 
We could also keep the in validationConfig format as a convention (you can take an inspiration from `ovirt-provider-tab-validation`)

suomiy replied:
also the check and save button should reflect the error state of this validation

glekner commented:
<img width="679" alt="Screen Shot 2020-08-11 at 11 52 17" src="https://user-images.githubusercontent.com/24938324/89877801-4a51b680-dbc9-11ea-818b-4e68c1f67d49.png">

@suomiy 
Added `validateVMwareSettings` function.
also added disable for Check and save button in case of invalid input.

suomiy commented on frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx:
```diff
@@ -96,6 +97,12 @@ const stateToProps = (state, { wizardReduxID, provider }) => {
     );
   }
 
+  if (provider === VMImportProvider.VMWARE && hasAllPrerequisiteValuesFiled) {
```
we don't have to do the test second time. You can just check if the validation is Error type.And you can do this for both providers with iGetProviderFieldAttribute

suomiy commented:
> Bug 1863000 should be left for the controller only.

let this bug for the UI and created a new one for the controller https://bugzilla.redhat.com/show_bug.cgi?id=1867980

yaacov commented:
/retest

suomiy commented on frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx:
```diff
@@ -86,6 +87,14 @@ const stateToProps = (state, { wizardReduxID, provider }) => {
       OvirtProviderField.USERNAME,
       VMWareProviderField.USERNAME,
+    ) &&
+    !iGetProviderFieldAttribute(
```
we have to check the validation type. It should fail for error/trivial error but pass for warn/info

suomiy commented on frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/import-providers-tab/providers/vm-import-password.tsx:
```diff
@@ -70,6 +72,17 @@ const stateToProps = (state, { wizardReduxID, provider }) => {
+    VMWareProviderField.HOSTNAME,
+  );
+  const isValidationCritical =
+    iGet(validationField, 'type') === ValidationErrorType.Error ||
```
let's use `[].includes` here for brevity

glekner replied:
fixed

suomiy commented:
/lgtm
